source/core/token-reader.h

```diff
@@ -0,0 +1,258 @@
+#ifndef CORE_TOKEN_READER_H
+#define CORE_TOKEN_READER_H
+
+#include "basic.h"
```

Comment:
I’m not exactly thrilled to see more of `CoreLib` make its way back in (I’ve already dealt with many “why in the world do you use your own string type?” questions. Do we really need a general-purpose lexer? Can this just be part of the render-test application instead?

Reply:
Not sure about this. But general tokenizing turns out to be a frequently used feature throughout the codebases of my other projects, so it might be useful to leave them in corelib.